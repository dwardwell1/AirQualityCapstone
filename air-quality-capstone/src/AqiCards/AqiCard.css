.jumbotron {
    border:rgba(255, 255, 255, 1) solid 1px;;
    border-radius: 25px;
    padding: 20px;
    width: 200px;
     background-color: rgba(255, 255, 255, 0.2);
 }
 