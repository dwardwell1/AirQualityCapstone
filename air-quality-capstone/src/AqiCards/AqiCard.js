import React, { useState, useEffect } from 'react';
import AqApi from '../api/aqApi';
import CardLoad from '../common/CardLoad';
import './AqiCard.css';

/** AQI Detail page.
 *
 * Renders information about zip codes API,
 *
 */

function AqiCard({ zip }) {
	console.debug('AqiDetail', 'zip=', zip);

	const [ aqi, setAqi ] = useState(null);

	useEffect(
		function getAqiForZip() {
			async function getAqi() {
				try {
					setAqi(await AqApi.getAqi(zip));
				} catch (err) {
					console.error(err);
				}
			}

			getAqi();
		},
		[ zip ]
	);

	if (!aqi) return <CardLoad />;

	return (
		<div className="jumbotron jumbtron-fluid mx-auto " style={{ width: '700px' }}>
			{console.log('aqi', aqi)}
			{aqi[1] ? (
				<div className="">
					<div className="container" className="mx-auto">
						<h2>Reporting Location: {aqi[1].ReportingArea}</h2>
						<hr className="mx-auto" />
						<h4 className="mx-auto">Current AQI: {aqi[1].AQI}</h4>
						<h4 className="mx-auto">How's my Air Quality?: {aqi[1].Category.Name}</h4>
					</div>
				</div>
			) : (
				<h4>No data for this zipcode</h4>
			)}
		</div>
	);
}

export default AqiCard;
