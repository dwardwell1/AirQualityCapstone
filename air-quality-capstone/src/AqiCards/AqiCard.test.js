import React from 'react';
import { render } from '@testing-library/react';
import AqiCard from './AqiCard';
import { MemoryRouter } from 'react-router';
//How to test final result and not loading spinner?

it('matches snapshot with zip', async function() {
	const { asFragment } = await render(
		<MemoryRouter>
			<AqiCard zip="95926" />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

it('Show loading spinner if loading', async function() {
	const { asFragment } = await render(
		<MemoryRouter>
			<AqiCard zip="xxxxx" />
		</MemoryRouter>
	);

	expect(asFragment()).toMatchSnapshot();
});
