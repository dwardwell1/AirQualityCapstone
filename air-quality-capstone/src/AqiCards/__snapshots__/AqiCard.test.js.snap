// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show loading spinner if loading 1`] = `
<DocumentFragment>
  <div
    class="text-center mt-5"
  >
    <div
      class="spinner-border"
      style="width: 3rem; height: 3rem;"
    >
      <span
        class="sr-only"
      >
        Air(Q)
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot with logo 1`] = `
<DocumentFragment>
  <div
    class="text-center mt-5"
  >
    <div
      class="spinner-border"
      style="width: 3rem; height: 3rem;"
    >
      <span
        class="sr-only"
      >
        Air(Q)
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot with zip 1`] = `
<DocumentFragment>
  <div
    class="text-center mt-5"
  >
    <div
      class="spinner-border"
      style="width: 3rem; height: 3rem;"
    >
      <span
        class="sr-only"
      >
        Air(Q)
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot without logo 1`] = `
<DocumentFragment>
  <div
    class="text-center mt-5"
  >
    <div
      class="spinner-border"
      style="width: 3rem; height: 3rem;"
    >
      <span
        class="sr-only"
      >
        Air(Q)
      </span>
    </div>
  </div>
</DocumentFragment>
`;
